File: bestia-models/src/main/java/bestia/model/domain/StatusEffect.java
package bestia.model.domain;

import java.io.Serializable;

import javax.persistence.Entity;
import javax.persistence.Id;

@Entity
public class StatusEffect implements Serializable {

	private static final long serialVersionUID = 1L;

	@Id
	private int id;

	private String datebaseName;

	/**
	 * Flag if the status is transmitted towards a client.
	 */
	private boolean isClientVisible;

	public int getId() {
		return id;
	}

	public String getDatebaseName() {
		return datebaseName;
	}

	public void setDatebaseName(String datebaseName) {
		this.datebaseName = datebaseName;
	}

	/**
	 * Flag if the status effect is transmitted to the client and shown inside
	 * the client GUI.
	 * 
	 * @return TRUE if the status effect should be visible to the client.
	 */
	public boolean isClientVisible() {
		return isClientVisible;
	}

}

File: bestia-models/src/main/kotlin/net/bestia/model/bestia/StatusEffect.kt
package net.bestia.model.bestia

import jakarta.persistence.Entity
import net.bestia.model.AbstractEntity

@Entity
data class StatusEffect(
    val datebaseName: String,
    /**
     * Flag if the status effect is transmitted to the client and shown inside
     * the client GUI. This is not the case for every status effect. If for example an equipment
     * modifies the bestia it will not be transferred to the client.
     *
     * @return TRUE if the status effect should be visible to the client.
     */
    val isClientVisible: Boolean = true
) : AbstractEntity()
